cache/ttl/ttl.go

```diff
@@ -0,0 +1,162 @@
+package ttl
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/renproject/kv/db"
+)
+
+var (
+	// PrunePointerTableName is the name of the table where we store the prune pointer.
+	PrunePointerTableName = "ttl_0"
+
+	// PrunePointerKey is the key of the key-value pair which we can use to
+	// query the current prune pointer.
+	PrunePointerKey = "prune_pointer"
+)
+
+type inMemTTL struct {
+	timeToLive    time.Duration
+	pruneInterval time.Duration
+	codec         db.Codec
+	db            db.DB
+}
+
+// New returns a new ttl wrapper over the given database.
+// The underlying database cannot have any database has a prefix of `ttl_`.
+func New(ctx context.Context, database db.DB, timeToLive time.Duration, pruneInterval time.Duration, codec db.Codec) db.DB {
+	ttlDB := inMemTTL{
+		timeToLive:    timeToLive,
+		pruneInterval: pruneInterval,
+		codec:         codec,
+		db:            database,
+	}
+
+	ttlDB.prune(ctx)
```

Comment:
It is preferable to not hide goroutines when possible. Instead, `prune` should be synchronous and it should be called with `go prune(ctx)`. This makes it clearer to the reader that `prune` is a long-lived background goroutine.